### PR TITLE
feat: middle out merge

### PR DESCRIPTION
**Pull request by spiermar**
Looking to implement middle-out merge, according to [this blog post](https://netflixtechblog.com/saving-13-million-computational-minutes-per-day-with-flame-graphs-d95633b6d01f). In a nutshell, it should filter out all stack frames until a string match is found (I'll likely change this to a regex before merging) in the "child" frame, basically ignoring all frames until the parent of a match. If there's no match in the whole stack, value goes to root.
Not the cleanest solution, but simplest for now. Let me know if you have a better idea.

### CODE REVIEW

File: python/nflxprofile/flamegraph.py [hunk 1]
```diff
@@ -193,10 +194,25 @@ def process(self, stack):
...
-        for frame in stack:
+        middle_out_filter = True
+        for i, frame in enumerate(stack):
             frame, frame_extras = self.process_frame(frame)
             if self.should_skip_frame(frame, frame_extras):
                 continue
+
+            if self.middle_out and middle_out_filter:
+                # middle out merge and no stack match yet
+                next_frame = stack[i + 1]
```
Comment (jasonk000) [inline python/nflxprofile/flamegraph.py]: What if `stack[i + 1]` is at top? Do we need a guard?

**Comment by jasonk000** [comment]
On intent : the intent is to have a user be able to enter or select a piece of text or regex over the a frame name and then only frames _beginning from that point_ would be shown, did I understand correctly?

The inverse of this use case would also be needed I think: I pick frame doCommonOperation and want to know who the callers are. Do we need an additional "direction"-like parameter that does the inverse of middle_out (ie: starts with it false and toggles to true?)

Thoughts?

File: python/nflxprofile/flamegraph.py (hunk 1 above)
Comment (spiermar) [reply python/nflxprofile/flamegraph.py]: True, I'll fix

**Comment by spiermar** [comment]
@jasonk000 Yes, that's the intent, but beginning from the parent. Inverted option, with middle-out should accomplish that the direction use case.

**Review by jasonk000 — APPROVED** [review]

**Pull request merged**